### PR TITLE
Api submission

### PR DESCRIPTION
**Pull request by H3ll3m4**
[Trello](https://trello.com/c/BdGST0MB/3393-api-submission-submitter-update)

## Goal
We want to replicate the feature from formbuilder into mojforms, and this is part of the api submission where we send json information to an endpoint url. We are getting the url and the key from the runner.

## Implementation
In the process of submission of a job, we added the action json, in the same way we treat csv and emails submissions.
We have to update the json schemas to accept this new action. we also added the required field kind to the schemas.

## Testing
Unit tests updated, but acceptances test remained on trello [ticket](https://trello.com/c/gjuaAuBf/3396-api-submission-acceptance-tests)
Acceptances tests were not broken on [deploy](https://app.circleci.com/pipelines/github/ministryofjustice/fb-submitter/3315/workflows/65becc29-98ac-42a1-914e-24a250020345/jobs/9129) 
Manual testing of attachments was performed and they have a mimetype, a filename, a pre-signed url for sharing, an encryption_key and encryption_iv.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -210,17 +211,17 @@ workflows:
...
-      - deploy_to_live_production:
-          context: *context
-          requires:
-            - acceptance_tests
-      - smoke_tests:
-          context: *context
-          requires:
-            - deploy_to_live_dev
-            - deploy_to_live_production
+      #       - deploy_to_test_production
```
Comment (StevenLeighton21) [inline .circleci/config.yml:214]: Need to undo this change before merge :)

File: app/services/adapters/jwe_webhook_destination.rb [hunk 2]
```diff
@@ -29,6 +29,9 @@ def body_as_string(body)
     end
 
     def encrypted_body(body)
+      Rails.logger.debug '********************* DEBUG'
```
Comment (StevenLeighton21) [inline app/services/adapters/jwe_webhook_destination.rb:32]: Should remove the debug logs before merge (and possibly consider adding the key and url to the log filter?)

File: app/services/attachment_parser_service.rb [hunk 3]
```diff
@@ -1,17 +1,30 @@
...
   def execute
     attachments.map do |attachment|
-      Attachment.new(
-        url: attachment.fetch(:url, nil),
-        mimetype: attachment.fetch(:mimetype),
-        filename: attachment.fetch(:filename),
-        type: attachment.fetch(:type),
-        path: nil
-      )
+      if @v2submission
```
Comment (StevenLeighton21) [inline app/services/attachment_parser_service.rb:9]: this does the same thing regardless of the value of v2submission?

**Review by StevenLeighton21 — APPROVED** [review]
Looks good - it looks like you probably have a redundant if statement (even though the test makes sense as v2 submissions have a slightly different hash style, you create the same object in the same way and could probably just add the error rescue you need). Not too concerned if that changes or is merged as is though 👍

File: app/services/attachment_parser_service.rb (hunk 3 above)
Comment (H3ll3m4) [reply app/services/attachment_parser_service.rb:9]: Yes it is either fetching a symbol :url or a string 'url'. It fails to parse the attachments if it is being given a symbol instead of a string so we have both cases. We could convert symbol hash to json and then parse json to string but the aim being to send a json to an api, I thought I would do that this way. I don't have other ideas.
